polygeist/test/polygeist-opt/polygeist-to-llvm.mlir

```diff
@@ -543,7 +543,7 @@ func.func private @ptr2memref(%arg0: !llvm.ptr) -> memref<?xf32> {
 // CHECK-NEXT:      llvm.return %[[VAL_6]] : i64
 // CHECK-NEXT:    }
 
-func.func private @non_bare_due_to_layout(%arg0: memref<100xi64, #layout>, %arg1: index) -> i64 {
+func.func private @non_bare_due_to_layout(%arg0: memref<100xi64, #layout>, %arg1: index) -> i64 attributes {llvm.linkage = #llvm.linkage<private>} {
```

Comment:
Why is it not enough to mark the function private? Can we add that to `couldBreakABI()`?

Reply:
`func.func -> llvm.func` conversion takes that property to set linkage [see](https://github.com/intel/llvm/blob/1a12bea11114650019325c069beec2977966521d/mlir/lib/Conversion/FuncToLLVM/FuncToLLVM.cpp#L420)

Reply:
ok, then should we change to 
```suggestion
func.func @non_bare_due_to_layout(%arg0: memref<100xi64, #layout>, %arg1: index) -> i64 attributes {llvm.linkage = #llvm.linkage<private>} {
```

Reply:
MLIR symbols visibility is different from linkage ([see](https://mlir.llvm.org/docs/Dialects/LLVM/#linkage)). I don't think that `private` hurts. Visibility does not affect `func->llvm` conversion at all.

Reply:
what bug me is `non_bare_due_to_layout` and `view` are not consistent, any reason why?

Reply:
I see. I removed `private` from `view`, not intended. Will fix